File: staircase/staircase.js
class staircase {
    constructor(n){
        this.n=n;
    }
    solve(){
        if (this.n<0 ^ this.n>100){
            return;
        };
        let resolve ="";
        resolve=" ".repeat(this.n-1)+"#";
        for (let i=this.n-2;i>=0;i--){
            resolve +="\n" + " ".repeat(i) + "#".repeat(this.n-i)
        };
       return resolve
    }
}

const const_class= new staircase(6)
console.log(const_class.solve())

File: staircase/staircase.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	fmt.Println(solve(6))
}

func solve(n int) string {
	if n < 0 || n > 100 {
		return ""
	}
	resolve := strings.Repeat(" ", n-1) + "#"
	for i := 2; i < n+1; i++ {
		resolve += "\n" + strings.Repeat(" ", n-i) + strings.Repeat("#", i)
	}
	return resolve
}
